src/features/calendar/components/index.tsx

```diff
@@ -13,12 +14,62 @@ export enum TimeScale {
   MONTH = 'month',
 }
 
+function isValidDate(date: Date) {
+  return date instanceof Date && !isNaN(date as unknown as number);
+}
```

Comment:
This expression stands out to me. According to the type on line 17, `date` is always a `Date`, so there should be no need to check the type with `instance of`, and casting it as `number` should not work.

The condition also seems unusual to me, because the second expression will only be evaluated if the first one is truthy. This means that this code relies on `date` being a `Date` and a `number` at the same time.

Can you elaborate?

Reply:
Yes, agree that it looks weird, and I should have written a comment at least. Since the query param is a string, I wanted to protect against invalid values, and passing an invalid date to isNaN returns true. Here's the Stack Overflow thread I got it from: https://stackoverflow.com/questions/1353684/detecting-an-invalid-date-date-instance-in-javascript

Reply:
Fair enough! Then I guess the only issue is the typing. Would this work?

```suggestion
function isValidDate(date: Date | unknown) {
  return date instanceof Date && !isNaN(date.getTime());
}
```

Reply:
Yes, that is clearer!